main.cpp

```diff
@@ -91,15 +91,118 @@ good to go!
  Wait for my code review.
  */
 
+
 #include <iostream>
-int main() 
+#include <math.h> 
+
+struct FloatType
+{
+    float add(float, float);
+    float subtract(float, float);
+    float multiply(float, float);
+    float divide(float, float);
+};
+    
+float FloatType::add(float lhs, float rhs) 
+{
+     return lhs + rhs; 
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;    
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs )
+{
+    if( rhs == 0.f )  
+    { 
+        printf("\nwarning, floating point division by zero returns 'inf' !\n");
+        return lhs/rhs;
+    }
+    
+    return lhs / rhs;
+}
+
+struct DoubleType
 {
+    double add(double, double);
+    double subtract(double, double);
+    double multiply(double, double);
+    double divide(double, double);
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs; 
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{ 
+    return lhs - rhs; 
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs; 
+}
+double DoubleType::divide(double lhs, double rhs)
+{
+    int zero = 0;
+    if(fabs(rhs - zero) < 1e-9 ) 
+    {   
+        printf("\nwarning, floating point division by zero returns 'inf' !\n");
+    }
+    
+    return lhs / rhs; 
+}
```

Comment:
don't check if `rhs` is 0 like this. 
write `if( rhs == 0.0 )` instead.  
don't use printf; use std::cout

- [ ] revise